### PR TITLE
Update transaction sign command to allow for incremental signing

### PR DESCRIPTION
**Pull request by Jimbo4350**
You can now specify an already signed transaction via `--tx-file` in order to add more signatures to it:  

`cardano-cli transaction sign (--tx-body-file FILE | --tx-file FILE) ...`

Resolves: #3545

### CODE REVIEW

**Review by newhoggy — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Types.hs [hunk 1]
```diff
@@ -241,3 +247,15 @@ data EpochLeadershipSchedule
...
 
+newtype TxBodyFile
+  = TxBodyFile FilePath
+  deriving Show
+
+newtype TxFile
+  = TxFile FilePath
+  deriving Show
+
+data TxBodyOrTxFile
```
Comment (cblp) [inline cardano-cli/src/Cardano/CLI/Types.hs:258]: InputTxFile already exists

**Review by cblp — CHANGES_REQUESTED** [review]

**Review by cblp — APPROVED** [review]

**Pull request merged**